test/functional/p2p_orphan_handling.py

```diff
@@ -0,0 +1,439 @@
+        peer.sync_with_ping()
+        assert_equal(len(peer.last_message["getdata"].inv), 2)
+        peer.wait_for_parent_requests([int(txid_conf_old, 16), int(missing_tx["txid"], 16)])
+
+        # Even though the peer would send a notfound for the "old" confirmed transaction, the node
+        # doesn't give up on the orphan. Once all of the missing parents are received, it should be
+        # submitted to mempool.
+        peer.send_message(msg_notfound(vec=[CInv(MSG_WITNESS_TX, int(txid_conf_old, 16))]))
+        peer.send_and_ping(msg_tx(missing_tx["tx"]))
+        peer.sync_with_ping()
+        assert_equal(node.getmempoolentry(orphan["txid"])["ancestorcount"], 3)
+
+    @cleanup
+    def test_orphans_overlapping_parents(self):
+        node = self.nodes[0]
+        # In the process of relaying inflight_parent_AB
+        peer_txrequest = node.add_p2p_connection(PeerTxRelayer())
+        # Sends the orphans
+        peer_orphans = node.add_p2p_connection(PeerTxRelayer())
+
+        self.log.info("Test handling of multiple orphans with missing parents that are already being requested")
+        # Parent of child_A only
+        missing_parent_A = self.wallet_nonsegwit.create_self_transfer()
+        # Parents of child_A and child_B
+        missing_parent_AB = self.wallet_nonsegwit.create_self_transfer()
+        inflight_parent_AB = self.wallet_nonsegwit.create_self_transfer()
+        # Parent of child_B only
+        missing_parent_B = self.wallet_nonsegwit.create_self_transfer()
+        child_A = self.wallet_nonsegwit.create_self_transfer_multi(
+            utxos_to_spend=[missing_parent_A["new_utxo"], missing_parent_AB["new_utxo"], inflight_parent_AB["new_utxo"]]
+        )
+        child_B = self.wallet_nonsegwit.create_self_transfer_multi(
+            utxos_to_spend=[missing_parent_B["new_utxo"], missing_parent_AB["new_utxo"], inflight_parent_AB["new_utxo"]]
+        )
+
+        # The wtxid and txid need to be the same for the node to recognize that the missing input
+        # and in-flight request for inflight_parent_AB are the same transaction.
+        assert_equal(inflight_parent_AB["txid"], inflight_parent_AB["tx"].getwtxid())
+
+        # Announce inflight_parent_AB and wait for getdata
+        peer_txrequest.send_and_ping(msg_inv([CInv(t=MSG_WTX, h=int(inflight_parent_AB["tx"].getwtxid(), 16))]))
+        self.fastforward(NONPREF_PEER_TX_DELAY)
+        peer_txrequest.wait_for_getdata([int(inflight_parent_AB["tx"].getwtxid(), 16)])
+
+        self.log.info("The node should not request a parent if it has an in-flight txrequest")
+        # Relay orphan child_A
+        self.relay_transaction(peer_orphans, child_A["tx"])
+        self.fastforward(NONPREF_PEER_TX_DELAY + TXID_RELAY_DELAY)
+        # Both missing parents should be requested.
+        peer_orphans.wait_for_parent_requests([int(missing_parent_A["txid"], 16), int(missing_parent_AB["txid"], 16)])
```

Comment:
"The node should not request a parent if it has an in-flight txrequest" ? Seems like parents are being reqiested. Maybe a typo or I can't tell what scenario it's covering.